src/ParallelAlgorithms/LinearSolver/Schwarz/OverlapHelpers.hpp

```diff
@@ -90,4 +97,175 @@ size_t overlap_num_points(const Index<Dim>& volume_extents,
 double overlap_width(size_t overlap_extent,
                      const DataVector& collocation_points) noexcept;
 
+/// Iterate over grid points in a region that extends partially into the volume
+class OverlapIterator {
+ public:
+  template <size_t Dim>
+  OverlapIterator(const Index<Dim>& volume_extents, size_t overlap_extent,
+                  const Direction<Dim>& direction) noexcept;
+
+  explicit operator bool() const noexcept;
```

Comment:
This conversion to bool for determining the end of iteration isn't really conventional for STL iterators. I don't mind it; it makes the iteration interface reasonably nice, but I definitely think it should be documented what's going on, and ideally a small snippet in the dox for how it should be used in a for loop.

Reply:
It's what `DataStructures/SliceIterator.hpp` does, so I went with the same approach.